src/Testcontainers/Configurations/WaitStrategies/UntilContainerIsHealthy.cs

```diff
@@ -0,0 +1,36 @@
+namespace DotNet.Testcontainers.Configurations.WaitStrategies
+{
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Containers;
+  using Microsoft.Extensions.Logging;
+
+  internal class UntilContainerIsHealthy : IWaitUntil
+  {
+    private readonly int failingStreak;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="UntilContainerIsHealthy"/> class.
+    /// </summary>
+    /// <param name="failingStreak">Number of tolerated failed attempts before throwing a <see cref="ContainerDidNotStartException"/>.</param>
+    public UntilContainerIsHealthy(int failingStreak)
+    {
+      this.failingStreak = failingStreak;
+    }
+
+    public Task<bool> Until(ITestcontainersContainer testcontainers, ILogger logger)
+    {
+      var health = testcontainers.Health;
+      if (testcontainers.State != TestcontainersStates.Running || health == null)
+      {
+        return Task.FromResult(false);
+      }
+
+      if (health.FailingStreak > this.failingStreak)
```

Comment:
I am afraid this won't work without further refactoring - the execution will fail at the first `Until` and won't reach the health check. Your idea was to perform the check the same way as in `UntilOperationIsSucceeded` stragegy, right? That would work if the `operation` was not `Func<bool>` but `Func<ITestcontainersContainer, bool>` instead since I need to pass context to the check function.

Reply:
> Your idea was to perform the check the same way as in UntilOperationIsSucceeded stragegy, right?

Ok, I do not remember anymore. I think I tested it.

Reply:
Got it now. Thanks!